File: OrchidCore/src/main/java/com/eden/orchid/impl/themes/components/ReadmeComponent.java
package com.eden.orchid.impl.themes.components;

import com.eden.orchid.api.OrchidContext;
import com.eden.orchid.api.resources.resource.OrchidResource;
import com.eden.orchid.api.theme.components.OrchidComponent;

import javax.inject.Inject;

public final class ReadmeComponent extends OrchidComponent {

    @Inject
    public ReadmeComponent(OrchidContext context) {
        super(context, "readme", 50);
    }

    public String getContent() {
        OrchidResource readmeResource = context.findClosestFile("readme");
        if (readmeResource != null) {
            return readmeResource.compileContent(this);
        }
        return null;
    }
}

File: OrchidCore/src/main/kotlin/com/eden/orchid/impl/themes/components/ReadmeComponent.kt
package com.eden.orchid.impl.themes.components

import com.eden.orchid.api.options.annotations.Description
import com.eden.orchid.api.resources.resourcesource.LocalResourceSource
import com.eden.orchid.api.resources.resourcesource.flexible
import com.eden.orchid.api.theme.components.OrchidComponent
import com.eden.orchid.utilities.SuppressedWarnings

@Description(value = "Locate and display your project's Readme file.", name = "Readme")
@Suppress(SuppressedWarnings.DEPRECATION)
class ReadmeComponent : OrchidComponent("readme") {
    val content: String?
        get() = context
            .getFlexibleResourceSource(LocalResourceSource, null)
            .findClosestFile(context, "readme")
            ?.compileContent(context, this)
}
